aws_xray_sdk/core/recorder.py

```diff
@@ -296,15 +296,46 @@ def begin_subsegment(self, name, namespace='local'):
             log.warning("No segment found, cannot begin subsegment %s." % name)
             return None
 
-        if not segment.sampled:
+        current_entity = self.get_trace_entity()
+
+        if not current_entity.sampled:
             subsegment = DummySubsegment(segment, name)
+            subsegment.sampled = False
         else:
             subsegment = Subsegment(name, namespace, segment)
 
         self.context.put_subsegment(subsegment)
 
         return subsegment
 
+    def begin_subsegment_without_sampling(self, name):
```

Comment:
There's lot of duplicate code here with `begin_subsegment`. Can you move the common code to another function and reuse?